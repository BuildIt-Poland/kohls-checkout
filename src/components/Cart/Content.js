import styled from 'styled-components';

import { spacingLarge } from '../../styles/designTokens';
import BaseContent from '../Content';

const Content = styled(BaseContent)`
  padding-top: ${spacingLarge};
`;

export default Content;
