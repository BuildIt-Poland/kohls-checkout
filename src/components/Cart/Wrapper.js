import styled from 'styled-components';

const Wrapper = styled.article`
  padding-bottom: 12rem;
`;

export default Wrapper;
