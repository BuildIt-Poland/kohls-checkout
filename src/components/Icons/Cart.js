import React from 'react';

function Cart() {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" aria-labelledby="icon-cart-title" width="41" height="37">
      <title id="icon-cart-title">Cart</title>
      <path
        fill="#222"
        d="M16.734 31.968a4.5 4.5 0 1 1-9 0 4.5 4.5 0 0 1 9 0zm10.103-4.5a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9zM9.422 5.598V.532H.837v3.377H6.03V23.69h27.076L40.36 5.598H9.422z"
      />
    </svg>
  );
}

export default Cart;
