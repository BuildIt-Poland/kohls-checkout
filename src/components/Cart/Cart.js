import React from 'react';
import Item from '../Item';

import { CHECKOUT_DELIVERY_PATH } from '../../constants/routes';
import Headline from '../Headline';
import NextStep from '../NextStep';
import OrderSummary from '../OrderSummary';
import Content from './Content';
import Wraper from './Wrapper';

function Cart() {
  return (
    <Wraper>
      <Content as="section">
        <Headline>Cart</Headline>
        <Item />
      </Content>
      <OrderSummary />
      <NextStep label="Checkout" to={CHECKOUT_DELIVERY_PATH} />
    </Wraper>
  );
}

export default Cart;
