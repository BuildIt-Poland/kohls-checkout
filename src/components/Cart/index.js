import { connect } from 'react-redux';

import { refillDemoCart } from '../../actions/cartActions';
import Cart from './Cart';

function mapStateToProps(state) {
  return {
    items: state.cart.items
  };
}

function mapDisaptchToProps(dispatch) {
  return {
    refillDemoCart: () => dispatch(refillDemoCart())
  };
}

export default connect(
  mapStateToProps,
  mapDisaptchToProps
)(Cart);
