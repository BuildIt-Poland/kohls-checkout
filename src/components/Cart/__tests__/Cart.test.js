import React from 'react';
import { create } from 'react-test-renderer';

import Cart from '../Cart';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - Cart', () => {
  it('renders Cart component', () => {
    const component = create(<Cart />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
