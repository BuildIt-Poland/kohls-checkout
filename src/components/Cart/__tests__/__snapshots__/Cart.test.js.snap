// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Cart renders Cart component 1`] = `
.c0 {
  padding-bottom: 12rem;
}

.c2 {
  margin: 0;
  margin-bottom: 2rem;
  font-size: 3rem;
  font-weight: 700;
  line-height: 1;
}

.c4 {
  background: #00a09f;
  color: #ffffff;
  display: block;
  font-size: 2.4rem;
  font-weight: 700;
  padding: 1.2rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 96rem;
  margin: 0 auto;
}

.c3 {
  bottom: 0;
  left: 0;
  padding: 1rem;
  position: fixed;
  width: 100%;
  background: #ffffff;
  border-top: 1px solid #b0b0b0;
  z-index: 6000;
}

.c1 {
  max-width: 100rem;
  padding: 0 2rem;
  margin: 0 auto;
}

<article
  className="c0"
>
  <InfoBanner>
    you have received 
    <strong>
      free shipping
    </strong>
  </InfoBanner>
  <section
    className="c1"
  >
    <h1
      className="c2"
    >
      Cart
    </h1>
    <ItemsList />
  </section>
  <OrderSummary />
  <div
    className="c3"
  >
    <Link
      className="c4"
      data-testid="next-step-button"
      to="/checkout/delivery"
    >
      Checkout
    </Link>
  </div>
</article>
`;
